src/services/paras/ParasService.ts

```diff
@@ -369,13 +391,20 @@ export class ParasService extends AbstractService {
 
 	/**
 	 * Calculate the current lease period index.
+	 * Ref: https://github.com/paritytech/polkadot/pull/3980
 	 *
-	 * @param blockHeight current blockheight
-	 * @param leasePeriod duration of lease period
+	 * @param historicApi
+	 * @param now Current block number
 	 */
-	private leasePeriodIndexAt(now: BN): BN {
-		const leasePeriod = this.api.consts.slots.leasePeriod as BlockNumber;
-		return now.div(leasePeriod);
+	private leasePeriodIndexAt(
+		historicApi: ApiDecoration<'promise'>,
+		now: BN
+	): BN {
+		const leasePeriod = historicApi.consts.slots.leasePeriod as BlockNumber;
+		const offset =
+			(historicApi.consts.slots.leaseOffset as BlockNumber) || BN_ZERO;
+
+		return now.sub(offset).div(leasePeriod);
```

Comment:
This needs to be updated slightly. There is an edge case where offset > now and we decided to consider this has have _no_ lease period index. In the substrate code this is represented by None. Here is the line https://github.com/paritytech/polkadot/pull/3980/commits/3668966dc02ac793c799d8c8667e8c396d891734#diff-3e19d8a242311f202d80bcf9998ffdc22ddd933ed340f308276f353f137e2304R444

So basically it says "if now - offset is less than 0, return None". So we need a update for the logic that uses this function to basically say there is no lease period at `now` and we could add the first lease period start at block `offset`. 

 (This logic which got added to the substrate PR a bit late).

Reply:
(You can see the changes for this in substrate commit 3668966dc02ac793c799d8c8667e8c396d891734)